tests/test02/structsinpkg/user_validator.go

```diff
@@ -7,15 +7,15 @@ import (
 func UserValidate(obj *User) []error {
 	var errs []error
 
-	if obj.FirstName == "" {
+	if !(obj.FirstName != "") {
```

Comment:
Eu não entendi bem o motivo da mudança. Pra mim ficou mais complicado de entender :)

Reply:
Vou tentar explicar mais. Antes, quero reforçar que este código em user_validator.go é gerado, então não precisa ser "bonito", mas funcional. Se vocês olharem o código gerado pelo protoc vão ver muitas bizarrices, mas funciona rsrsrs.

O código gerado anteriormente era sim "mais óbvio", mas para gerá-lo o gerador tinha que usar "lógica invertida" em alguns casos (exemplo abaixo), o que deixa o código mais difícil de gerar, entender e manter.

Por exemplo, para a tag LTE (less than or equal), para o tipo uint8, tinha uma entrada no map como esta:

```"lte,uint8":       {"{{.Name}}", ">", `{{.Target}}`, "{{.Name}} must be <= {{.Target}}"}```

Notem o operator ">", apesar da tag ser LTE, o que sugere um "<=" e não ">".

Invertendo com o not no código gerado, a entrada no map passa a ser "clara":

```"lte,uint8":       {"{{.Name}}", "<=", `{{.Target}}`, "{{.Name}} must be <= {{.Target}}"}```

Com isso fica mais fácil criar as outras tags e variações (LT, GT, GTE, etc) para todos os tipos.

Ficou mais claro a intenção?